project_euler/problem_234/sol2.py

```diff
@@ -0,0 +1,124 @@
+"""
+Project Euler Problem 234: https://projecteuler.net/problem=234
+
+For an integer n ≥ 4, we define the lower prime square root of n, denoted by lps(n), as
+the largest prime ≤ √n and the upper prime square root of n, ups(n), as the smallest
+prime ≥ √n.
+
+So, for example, lps(4) = 2 = ups(4), lps(1000) = 31, ups(1000) = 37.
+Let us call an integer n ≥ 4 semidivisible, if one of lps(n) and ups(n) divides n,
+but not both.
+
+The sum of the semidivisible numbers not exceeding 15 is 30, the numbers are 8,
+10 and 12.
+15 is not semidivisible because it is a multiple of both lps(15) = 3 and ups(15) = 5.
+As a further example, the sum of the 92 semidivisible numbers up to 1000 is 34825.
+
+What is the sum of all semidivisible numbers not exceeding 999966663333?
+"""
+
+
+from __future__ import annotations
+
+import itertools
+from math import ceil
+from typing import Iterator
+
+
+def sieve(limit: int) -> Iterator[int]:
+    """
+    Return an iterable of all the primes less than or equal to limit in ascending order.
+    >>> list(sieve(5))
+    [2, 3, 5]
+    >>> list(sieve(50))
+    [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]
+    """
+    primes = [False for _ in range(limit + 1)]
+    primes[2] = True
+    primes[3] = True
+    for i in range(3, limit + 1, 2):
+        primes[i] = True
+    for i in range(3, limit + 1, 2):
+        if not primes[i]:
+            continue
+        for j in range(i * i, limit + 1, i):
+            primes[j] = False
+
+    return itertools.compress(itertools.count(), primes)
+
+
+def pairwise(iterable: Iterator[int]) -> Iterator[tuple]:
```

Comment:
```suggestion
def pairwise(iterable: Iterator[int]) -> Iterator[tuple[int]]:
```
Is that correct?
Also, do you want an `Iterator` or an `Iterable` as your input?

Reply:
Absolutey right, thanks!